scripts/lean-depot.py

```diff
@@ -0,0 +1,293 @@
+    pkgs = { os.path.splitext(fn)[0] : toml.load(f'pkgs/{fn}')['package']
+             for fn in os.listdir('pkgs')
+             if os.path.splitext(fn)[1] == '.toml' }
+    print(pkgs)
+
+    snap = { 'snapshot' : { k : { 'git' : pkg['url'],
+                                  'rev' : pkg['commit'],
+                                  'desc' : pkg['description'] }
+                                for (k,pkg) in pkgs.items() } }
+    with open(path, 'w') as h:
+        print(toml.dump(snap, h))
+
+@user_command('resolver')
+def resolver():
+    """find snapshots to accomodate all dependencies"""
+    prj   = toml.load('lean-depot.toml')
+    snap  = find_best_match(prj)
+    prj['package']['lib_snapshot'] = snap
+    with open('lean-depot.toml', 'w') as h:
+        h.write(toml.encoder.dumps(prj, toml.encoder.TomlPreserveInlineDictEncoder()))
+    configure()
+
+@user_command('list')
+def list_packages():
+    """list available packages"""
+    prj   = toml.load('lean-depot.toml')
+    tag   = prj['package']['lib_snapshot']
+    vers  = prj['package']['lean_version']
+    deps  = prj['dependencies']
+    depot = toml.load(get_depot(vers)[tag])
+    for (k,v) in depot['snapshot'].items():
+        print(k)
+        print(wrap_indent(2, v['desc'], 60))
+        print('')
+
+@user_command('snapshot')
+def snapshot():
+    """list available snapshots"""
+    prj   = toml.load('lean-depot.toml')
+    vers  = prj['package']['lean_version']
+    depot = get_depot(vers)
+    for (k,v) in depot.items():
+        print(k)
+        print(wrap_indent(2, ', '.join(toml.load(v)['snapshot'].keys()), 60))
+
+@user_command('nightly')
+def nightly():
+    """test package against nightly snapshot (not implemented)"""
+    print("not implemented")
+
+parser = argparse.ArgumentParser()
+parser.add_argument('cmd', choices=l.keys(), # required=False,
+                    help='command to get information on')
+
+@user_command('--help', parser)
+def help(cmd):
+    """provide this help screen"""
+    usage()
+
+def usage():
+    w = max([ len(i) for i in l.keys() ])
+    print(f'usage: {sys.argv[0]} [-h]')
+    print('')
+    print('Use currated set of Lean packages')
+    print('')
+    print('positional arguments:')
+    for (k,v) in l.items():
+        if k != '--help':
+            print(f'  {str.ljust(k,w+3)}{v[0].__doc__}')
+
+    print('')
+    print('optional arguments:')
+    print('  -h, --help  show this help message and exit')
+
+if __name__ == '__main__':
+    argv = sys.argv
+    if len(argv) > 1 and argv[1] in l:
+         (f,parser) = l[argv[1]]
+         if parser:
+             args = parser.parse_args(argv[2:])
+             f(**args.__dict__)
+         elif len(argv) == 2:
+             f()
+         else:
+             usage()
+    else:
+        usage()
+
+# Local Variables:
+# eval: (add-hook 'after-save-hook (lambda () (copy-file (buffer-file-name) "~/.mathlib/bin/lean-depot" t)) nil t)
```

Comment:
I think this will be annoying to other people editing the file. Why not use a symlink?

Reply:
Thanks for the suggestion